files/build_templates/sonic_debian_extension.j2

```diff
@@ -512,6 +514,10 @@ sudo LANG=C chroot $FILESYSTEM_ROOT docker $SONIC_NATIVE_DOCKERD_FOR_DOCKERFS ta
 {% endif %}
 {% endfor %}
 
+# Copy docker start script to be able to start docker in chroot
+sudo mkdir -p $FILESYSTEM_ROOT_USR_LIB/docker
+sudo cp $DOCKER_SCRIPTS_DIR/docker $FILESYSTEM_ROOT_USR_LIB/docker/docker.sh
```

Comment:
I notice it is already copied. Could you use this one?
```
sudo cp files/docker/docker $FILESYSTEM_ROOT/etc/init.d/
```

Reply:
It is later removed by:
```
sudo rm $FILESYSTEM_ROOT/etc/init.d/docker
```
And I don't want to put it in /etc/init.d/ as this is not expected to be used as init.d script, docker is a systemd service.

Reply:
Thanks for the insight. That change is introduced by [https://github.com/Azure/sonic-buildimage/pull/2417/files](https://github.com/Azure/sonic-buildimage/pull/2417/files)

The purpose of the that PR and this PR is similar, and I see that PR overwrite the vanilla file and later remove it, so the vanilla `/etc/init.d/docker` file is missing. I suggest unify the 2 effort and I agree `/etc/init.d/docker` may be not a good place.

Your argument "docker is a systemd service" does not relating because you are not using `systemctl` to start service in your use case
```
root@sonic:~$ /usr/lib/docker/docker.sh start
```


---
In reply to: [570580898](https://github.com/Azure/sonic-buildimage/pull/5935#discussion_r570580898) [](ancestors = 570580898)

Reply:
Thanks, I see what you mean. I made a change to copy it once and not remove.
My comment was an argument against /etc/init.d/, as later docker debian packages removed this script and putting it back to /etc/init.d/ gives SONiC users wrong idea that user do: "service docker restart" but I doubt that will work correctly, at least won't consider systemd dependencies, so I choose /usr/lib/docker to hide this script from regular user.